Configurations/make-release-package.sh

```diff
@@ -80,70 +104,60 @@ if [ "$ACTION" = "" ] ; then
+        find . \! -type d | rev | sort | rev | tar cv --files-from=- | xz -9 > "../Sparkle-$MARKETING_VERSION.tar.xz"
 
         # Copy archived distribution for CI
-        cp -f "../Sparkle-$CURRENT_PROJECT_VERSION.tar.xz" "../sparkle-dist.tar.xz"
+        cp -f "../Sparkle-$MARKETING_VERSION.tar.xz" "../sparkle-dist.tar.xz"
 
         # Extract archive for testing binary validity
-        tar -xf "../Sparkle-$CURRENT_PROJECT_VERSION.tar.xz" -C "/tmp/sparkle-extract"
+        tar -xf "../Sparkle-$MARKETING_VERSION.tar.xz" -C "/tmp/sparkle-extract"
     else
         # Fallback to bz2 compression if xz utility is not available
-        find . \! -type d | rev | sort | rev | tar cjvf "../Sparkle-$CURRENT_PROJECT_VERSION.tar.bz2" --files-from=-
+        find . \! -type d | rev | sort | rev | tar cjvf "../Sparkle-$MARKETING_VERSION.tar.bz2" --files-from=-
 
         # Extract archive for testing binary validity
-        tar -xf "../Sparkle-$CURRENT_PROJECT_VERSION.tar.bz2" -C "/tmp/sparkle-extract"
+        tar -xf "../Sparkle-$MARKETING_VERSION.tar.bz2" -C "/tmp/sparkle-extract"
     fi
 
     # Test code signing validity of the extracted products
     # This guards against our archives being corrupt / created incorrectly
-    codesign --verify -vvv --deep "/tmp/sparkle-extract/Sparkle.framework"
-    codesign --verify -vvv --deep "/tmp/sparkle-extract/XPCServices/org.sparkle-project.Downloader.xpc"
-    codesign --verify -vvv --deep "/tmp/sparkle-extract/XPCServices/org.sparkle-project.InstallerConnection.xpc"
-    codesign --verify -vvv --deep "/tmp/sparkle-extract/XPCServices/org.sparkle-project.InstallerLauncher.xpc"
-    codesign --verify -vvv --deep "/tmp/sparkle-extract/XPCServices/org.sparkle-project.InstallerStatus.xpc"
-    codesign --verify -vvv --deep "/tmp/sparkle-extract/sparkle.app"
-    codesign --verify -vvv --deep "/tmp/sparkle-extract/Sparkle Test App.app"
-    codesign --verify -vvv --deep "/tmp/sparkle-extract/bin/BinaryDelta"
-    codesign --verify -vvv --deep "/tmp/sparkle-extract/bin/generate_appcast"
-    codesign --verify -vvv --deep "/tmp/sparkle-extract/bin/sign_update"
-    codesign --verify -vvv --deep "/tmp/sparkle-extract/bin/generate_keys"
+    verify_code_signatures "/tmp/sparkle-extract"
 
     rm -rf "/tmp/sparkle-extract"
     rm -rf "$CONFIGURATION_BUILD_DIR/staging"
     
     # Generate zip containing the xcframework for SPM
     cd "$CONFIGURATION_BUILD_DIR/staging-spm"
-    #rm -rf "$CONFIGURATION_BUILD_DIR/Sparkle.xcarchive"
-    zip -rqyX -9 "../Sparkle-for-Swift-Package-Manager.zip" *
+    rm -rf "$CONFIGURATION_BUILD_DIR/Sparkle.xcarchive"
+    ditto -c -k --zlibCompressionLevel 9 --rsrc . "../Sparkle-for-Swift-Package-Manager.zip"
+
+    # Test code signing validity of the extracted Swift package
+    # This guards against our archives being corrupt / created incorrectly
+    verify_code_signatures "$CONFIGURATION_BUILD_DIR/staging-spm"
```

Comment:
We would want to verify the signatures of the extracted zip file, not the contents before being zipped. Should be a small change I think. To mirror above, I think this ought to do it:

```
ditto -x -k "../Sparkle-for-Swift-Package-Manager.zip" /tmp/sparkle-extract
verify_code_signatures "/tmp/sparkle-extract"

rm -rf "/tmp/sparkle-extract"
rm -rf "$CONFIGURATION_BUILD_DIR/staging-spm"

Reply:
Done in e87eda83d207cd4b7b2359fa04150f1239817730.

Reply:
Not quite, this snippet I wrote here is extracting the .zip into a temp directory and verifying the code signatures of the extraction. Current code is still verifying code signatures inside staging-spm instead.